Allow out-of-cluster operation of VPA

JensErat opened the pull request:
Because of how we operate our Kubernetes clusters, we need to run vertical pod autoscaler with an external kubeconfig. Also aligned and factored out redundant configuration code.

bskiba commented on vertical-pod-autoscaler/pkg/admission-controller/main.go:
```diff
@@ -58,6 +56,9 @@ var (
 
 	port            = flag.Int("port", 8000, "The port to listen on.")
 	address         = flag.String("address", ":8944", "The address to expose Prometheus metrics.")
+	kubeconfig      = flag.String("kubeconfig", "", "Paths to a kubeconfig. Only required if out-of-cluster.")
```
s/Paths/path/

JensErat replied:
I went for a capital `Path` for consistency, I guess you were only after the trailing `s` anyway.

bskiba replied:
Hm, I still see the s at the end?

bskiba replied:
Here and in recommender/main.go

bskiba commented on vertical-pod-autoscaler/pkg/updater/main.go:
```diff
@@ -72,16 +74,14 @@ func main() {
 	vpaClient := vpa_clientset.NewForConfigOrDie(config)
 	factory := informers.NewSharedInformerFactory(kubeClient, defaultResyncPeriod)
 	targetSelectorFetcher := target.NewVpaTargetSelectorFetcher(config, kubeClient, factory)
 	var limitRangeCalculator limitrange.LimitRangeCalculator
```
nit: You can remove this line now

bskiba commented on vertical-pod-autoscaler/common/kubeconfig.go:
```diff
@@ -0,0 +1,31 @@
+)
+
+func CreateKubeConfigOrDie(kubeconfig string, kubeApiQps float32, kubeApiBurst int) *rest.Config {
+	//var config *rest.Config
```
nit: please remove

mwielgus commented:
@JensErat 
Please address @bskiba comments.

JensErat commented:
Sorry for the delay, too much other work to do and some vacation in-between. Thank you for the review! Rebased, resolved minors and split up the commit as requested.

JensErat commented:
Looking at other PRs, it seems master is broken and not just flaky?

    FAIL	k8s.io/autoscaler/cluster-autoscaler/cloudprovider/aws [build failed]

/retest

JensErat commented:
Seems like #3222 will fix the broken test, will rebase as soon as merged.

JensErat commented:
@mwielgus @bskiba review comments should be fine now, and tests also work since AWS test is fixed in master.

bskiba commented on vertical-pod-autoscaler/common/kubeconfig.go:
```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
nit: s/2018/2020/

bskiba commented:
Two more small comments

mwielgus commented:
Where we are with this PR? It seems that it just needs two nit-fixes and a rebase...

mwielgus commented:
Inactivity warning. Please resolve the comments and fix conflicts.

mwielgus commented:
Closing due to inactivity. Feel free to reopen.

JensErat commented:
/reopen

I'm sorry for the long period of inactivity. I updated most of the minors and will comment on the remaining as soon as prow reopened and github updated my branch.
